component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/backupcode/BackupCodeAuthenticator.java

```diff
@@ -621,19 +610,10 @@ private boolean isValidBackupCode(String token, AuthenticationContext context, S
      * @throws BackupCodeException If an error occurred while removing the used backup code.
      */
     private void removeUsedBackupCode(String userToken, String username, List<String> backupCodes)
-            throws BackupCodeException {
+            throws BackupCodeException, NoSuchAlgorithmException {
```

Comment:
We do not need to throw `NoSuchAlgorithmException` from here. Ideally the util method should handle it.

Reply:
addressed with [7d9674b](https://github.com/wso2-extensions/identity-outbound-auth-backup-code/pull/6/commits/7d9674b6bf172b43c30479b0d64fdfa6f8130373)